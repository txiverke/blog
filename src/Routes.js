// @flow

import React from 'react'
import { Switch, Route } from 'react-router-dom'
import RoutesAsync from './RoutesAsync'

const Routes = () => (
  <Switch>
    <Route
      exact
      path="/"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/Landing')} />
      )} />
    <Route
      path="/about-me"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/About')} />
      )} />
    <Route
      path="/posts/:slug"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/PostView')} />
      )} />
    <Route
      path="/posts"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/Posts')} />
      )} />
    <Route
      path="/projects/:id"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/ProjectView')} />
      )} />
    <Route
      path="/projects"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/Projects')} />
      )} />
    <Route 
      path="/admin/about-me"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/AdminSections')} />
      )} />
    <Route 
      path="/admin/posts/:id"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/AdminPostEdit')} />
      )} />  
    <Route 
      path="/admin/posts"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/AdminSections')} />
      )} />
    <Route 
      path="/admin/projects"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/AdminSections')} />
      )} />
    <Route 
      path="/admin"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/Admin')} />
      )} />
    <Route
      path="/sign-in"
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/SignIn')} />
      )} />
    <Route
      component={props => (
        <RoutesAsync props={props} loadingPromise={import('./views/NotFound')} />
      )} />
  </Switch>
)

export default Routes