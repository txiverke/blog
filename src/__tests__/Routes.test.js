// @flow

import React from 'react'
import { shallow } from 'enzyme'
import Routes from '../Routes'

describe('<Routes />', () => {
  it('should render itself', () => {
    const component = shallow(<Routes />)
    expect(component).toMatchSnapshot()
  })
})