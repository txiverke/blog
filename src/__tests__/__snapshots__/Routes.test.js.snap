// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          path="/about-me"
/>,
        <Route
          component={[Function]}
          path="/posts/:slug"
/>,
        <Route
          component={[Function]}
          path="/posts"
/>,
        <Route
          component={[Function]}
          path="/projects/:id"
/>,
        <Route
          component={[Function]}
          path="/projects"
/>,
        <Route
          component={[Function]}
          path="/admin/about-me"
/>,
        <Route
          component={[Function]}
          path="/admin/posts/:id"
/>,
        <Route
          component={[Function]}
          path="/admin/posts"
/>,
        <Route
          component={[Function]}
          path="/admin/projects/:id"
/>,
        <Route
          component={[Function]}
          path="/admin/projects"
/>,
        <Route
          component={[Function]}
          path="/admin"
/>,
        <Route
          component={[Function]}
          path="/sign-in"
/>,
        <Route
          component={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/about-me",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/posts/:slug",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/posts",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/projects/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/projects",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin/about-me",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin/posts/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin/posts",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin/projects/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin/projects",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/sign-in",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            path="/about-me"
/>,
          <Route
            component={[Function]}
            path="/posts/:slug"
/>,
          <Route
            component={[Function]}
            path="/posts"
/>,
          <Route
            component={[Function]}
            path="/projects/:id"
/>,
          <Route
            component={[Function]}
            path="/projects"
/>,
          <Route
            component={[Function]}
            path="/admin/about-me"
/>,
          <Route
            component={[Function]}
            path="/admin/posts/:id"
/>,
          <Route
            component={[Function]}
            path="/admin/posts"
/>,
          <Route
            component={[Function]}
            path="/admin/projects/:id"
/>,
          <Route
            component={[Function]}
            path="/admin/projects"
/>,
          <Route
            component={[Function]}
            path="/admin"
/>,
          <Route
            component={[Function]}
            path="/sign-in"
/>,
          <Route
            component={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/about-me",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/posts/:slug",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/posts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/projects/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin/about-me",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin/posts/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin/posts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin/projects/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin/projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/sign-in",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
